R/tt_export.R

```diff
@@ -630,5 +427,517 @@ export_as_pdf <- function(tt,
+#' @export
+margins_potrait <- function() {
+  officer::page_mar(bottom = 0.98, top = 0.95, left = 1.5, right = 1, gutter = 0)
+}
+#' @describeIn export_as_docx helper function that defines standard landscape margins for tables.
+#' @export
+margins_landscape <- function() {
+  officer::page_mar(bottom = 1, top = 1.5, left = 0.98, right = 0.95, gutter = 0)
+}
+
+#' Create a `FlexTable` from an `rtables` table
+#'
+#' @description
+#' Principally used for export ([export_as_docx()]), this function produces a `flextable`
+#' from an `rtables` table. If `theme = NULL`, `rtables`-like style will be used. Otherwise,
+#' [theme_docx_default()] will produce a `.docx`-friendly table.
+#'
+#' @inheritParams gen_args
+#' @param theme function(1). Defaults to `theme_docx_default(tt)`. It expects a
+#'   a theme function that is designed internally as a function of a `flextable` object
+#'   and changes its layout and style. If set to `NULL`, it will produce a table similar
+#'   to `rtables` default.
+#' @param border `officer` border object. Defaults to `officer::fp_border(width = 0.5)`.
+#' @param indent_size integer(1). If `NULL`, the default indent size of the table (see
+#'   [matrix_form()] `indent_size`) is used. To work with `docx`, any size is multiplied
+#'   by 5 as default.
+#' @param titles_as_header logical(1). Defaults to `TRUE` and makes additional header rows
+#'   for [main_title()] string and [subtitles()] character vector (one per element). If `FALSE`
+#'   it is still possible to use the same parameter in [export_as_docx()] to add these titles
+#'   as a text paragraph above the table.
+#' @param paginate logical(1). If you need `.docx` export and you use
+#'   `export_as_docx`, we suggest relying on `word` pagination system. Cooperation
+#'   between the two mechanisms is not guaranteed. This option splits `tt` in different
+#'   "pages" as multiple `flextables`. Defaults to `FALSE`.
+#' @inheritParams paginate_table
+#' @param total_width numeric(1). Total width in inches for the resulting
+#'   `flextable(s)`. Defaults to 10.
+#'
+#' @return a `flextable` object.
+#'
+#' @seealso [export_as_docx()]
+#'
+#' @examples
+#' analysisfun <- function(x, ...) {
+#'   in_rows(
+#'     row1 = 5,
+#'     row2 = c(1, 2),
+#'     .row_footnotes = list(row1 = "row 1 - row footnote"),
+#'     .cell_footnotes = list(row2 = "row 2 - cell footnote")
+#'   )
+#' }
+#'
+#' lyt <- basic_table(
+#'   title = "Title says Whaaaat", subtitles = "Oh, ok.",
+#'   main_footer = "ha HA! Footer!"
+#' ) %>%
+#'   split_cols_by("ARM") %>%
+#'   analyze("AGE", afun = analysisfun)
+#'
+#' tbl <- build_table(lyt, ex_adsl)
+#' # rtables style
+#' tt_to_flextable(tbl, theme = NULL)
+#'
+#' tt_to_flextable(tbl, theme = theme_docx_default(tbl, font_size = 7))
+#'
+#' @name tt_to_flextable
+#' @export
+tt_to_flextable <- function(tt,
+                            theme = theme_docx_default(tt),
+                            border = officer::fp_border(width = 0.5),
```

Comment:
i have a bit mixed feeling about this. flextable is an object on its own, but we are mixing here with officer parameter, i would suggest to decouple the two

Reply:
it makes perfect sense! removed it ;)